src/lottery/assembly/fee-strategies.ts

```diff
@@ -1,39 +1,41 @@
-import { u128 } from "near-sdk-as";
+import { u128, env, logging } from "near-sdk-as";
 
 export enum StrategyType {
   Constant,
   Linear,
   Exponential,
 }
 
-export class Strategy {
+@nearBindgen
+export class FeeStrategy {
 
-  static selector(strategy: StrategyType, scalar: u32, base: u128): u128 {
-    switch (strategy) {
-      case StrategyType.Exponential:
-        return this.exponential(scalar, base);
+  strategy: StrategyType;
 
+  calculate_fee(scalar: u32, base: u128): u128 {
+    let fee: u128 = u128.Zero;
+    switch (this.strategy) {
+      case StrategyType.Exponential:
+        fee = this.calculate_exponential(scalar, base);
       case StrategyType.Linear:
-        return this.linear(scalar, base);
-
+        fee = this.calculate_linear(scalar, base);
       case StrategyType.Constant:
-        return this.constant(scalar, base);
-
+        fee = this.calculate_constant(scalar, base);
       default:
-        assert(false, "Must provide a valid fee strategy");
-        return u128.Zero;
+        logging.log("Unexpected StrategyType encountered");
```

Comment:
Until panic accepts a message, use logging.log and then panic, instead of assert(false,  "msg")